cleanhttp.go

```diff
@@ -26,3 +29,11 @@ func DefaultClient() *http.Client {
 		Transport: DefaultTransport(),
 	}
 }
+
+func SetFinalizer(transport *http.Transport) {
+	runtime.SetFinalizer(&transport, FinalizeTransport)
+}
+
+func FinalizeTransport(t **http.Transport) {
+	(*t).CloseIdleConnections()
```

Comment:
Mostly I didn't see a reason *not* to expose it. I can make it private. I'll test; if there are no wonky behaviors I'll inline it.